### PR TITLE
[DatePicker][RangeDatePicker]: added ability to enter date in format different than props.format #806 and #828

### PR DESCRIPTION
**Pull request by Sushasgit**
Info ℹ️ 
https://github.com/epam/UUI/issues/806
https://github.com/epam/UUI/issues/828

### CODE REVIEW

File: uui-components/src/inputs/DatePicker/BaseRangeDatePicker.tsx [hunk 1]
```diff
@@ -110,8 +116,8 @@ export abstract class BaseRangeDatePicker<TProps extends BaseRangeDatePickerProp
     }
 
     setValue = (value: PickerBodyValue<RangeDatePickerValue>) => {
-        this.props.value && (this.props.value.from !== value.selectedDate.from || this.props.value.to !== value.selectedDate.to) && this.handleValueChange(value.selectedDate);
-        this.setState({ ...this.state, inputValue: toCustomDateRangeFormat(value.selectedDate, this.getFormat()), ...value });
+        this.handleValueChange(value.selectedDate);
```
[the remark this answers is not in the document]
Comment (AlekseyManetov) [reply uui-components/src/inputs/DatePicker/BaseRangeDatePicker.tsx:119]: add  set selectedDate(check null value)

File: uui-components/src/inputs/DatePicker/BaseRangeDatePicker.tsx [hunk 2]
```diff
@@ -142,19 +148,24 @@ export abstract class BaseRangeDatePicker<TProps extends BaseRangeDatePickerProp
 
     onRangeChange = (value: PickerBodyValue<RangeDatePickerValue>) => {
         const isFromValueChanged = this.props.value.from != value.selectedDate.from;
+        const newValue = {
```
Comment (AlekseyManetov) [inline uui-components/src/inputs/DatePicker/BaseRangeDatePicker.tsx:151]: revert (?)

**Review by AlekseyManetov — CHANGES_REQUESTED** [review]

File: uui-components/src/inputs/DatePicker/BaseDatePicker.tsx [hunk 3]
```diff
@@ -77,16 +77,20 @@ export abstract class BaseDatePicker<TProps extends DatePickerCoreProps> extends
     handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
         if (isChildFocusable(e)) return;
         this.onToggle(false);
-        if (this.state.inputValue && !this.getIsValidDate(this.state.inputValue)) {
+        if (this.state.inputValue && this.getIsValidDate(this.state.inputValue)) {
```
[the remark this answers is not in the document]
Comment (Sushasgit) [reply uui-components/src/inputs/DatePicker/BaseDatePicker.tsx]: Thanks, done

File: uui-components/src/inputs/DatePicker/BaseRangeDatePicker.tsx (hunk 2 above)
Comment (Sushasgit) [reply uui-components/src/inputs/DatePicker/BaseRangeDatePicker.tsx]: Thanks, reverted

**Review by AlekseyManetov — CHANGES_REQUESTED** [review]

File: uui-components/src/inputs/DatePicker/BaseRangeDatePicker.tsx [hunk 4]
```diff
@@ -76,11 +76,11 @@ export abstract class BaseRangeDatePicker<TProps extends BaseRangeDatePickerProp
     }
 
     valueIsValid(value: string, inputType: RangeDatePickerInputType) {
-        if (dayjs(value, this.getFormat(), true).isValid()) {
+        if (dayjs(value, supportedDateFormats, true).isValid()) {
```
[the remark this answers is not in the document]
Comment (Sushasgit) [reply uui-components/src/inputs/DatePicker/BaseRangeDatePicker.tsx]: Thanks, done

**Pull request merged**